addons/account/models/account_invoice.py

```diff
@@ -388,6 +388,17 @@ def _get_vendor_display_info(self):
             invoice.vendor_display_name = vendor_display_name
             invoice.invoice_icon = invoice.source_email and '@' or ''
 
+    @api.multi
+    def _get_computed_reference(self):
+        if self.reference_type == 'invoice_number':
+            trailing_digits_str = re.match('.*?([0-9]+)$', self.number).group(1)
+            digits = len(trailing_digits_str)
+            trailing_digits = int(trailing_digits_str)
+            magical_modulo = 97
+            return '%s-%s' % (self.number, str(trailing_digits % magical_modulo).rjust(digits, '0'))
```

Comment:
I don't understand the use case: when would you want a communication that looks like: INV/2018/0001-0001? The last part will always be the same (we can't have INV/2018/0001-0002 for example), so why not use self.number instead? 
Edit: well actually we can have INV/2018/0098-0001 but I still fail to see how useful it is

Reply:
@csnauwaert when you pay partially you will have several payments for the same invoice, that's what I get here, a perfect use case if I understand well enough.